src/Reference/HasOne.php

```diff
@@ -164,8 +164,10 @@ public function init(): void
             $this->our_field = $this->link;
         }
 
-        if (!$this->owner->hasField($this->our_field)) {
-            $this->owner->addField($this->our_field, [
+        $ourModel = $this->getOurModel();
+
+        if (!$ourModel->hasField($this->our_field)) {
```

Comment:
what is this condition for?

Reply:
Only add field if not existing in model I guess

Reply:
i see "if not", probably right...